src/basic/hexdecoct.c

```diff
@@ -606,7 +606,7 @@ ssize_t base64mem_full(
         if (!r)
                 return -ENOMEM;
 
-        for (x = p; x < (const uint8_t*) p + (l / 3) * 3; x += 3) {
+        for (x = p; x && x < (const uint8_t*) p + (l / 3) * 3; x += 3) {
```

Comment:
I'm commenting here because it seems I can't on the binary file:
please shorten the name so that the test output width doesn't explode.

Reply:
OK. Done.

Reply:
Thanks. I'll merge this now, because I want to include those patches in stable backport.